src/main/scala/freeeventsourcing/syntax/ProcessSyntax.scala

```diff
@@ -0,0 +1,218 @@
+package freeeventsourcing.syntax
+
+import java.time.Instant
+import scala.annotation.implicitNotFound
+import cats.Monad
+import cats.free.Free
+import freeeventsourcing._
+import freeeventsourcing.EventSelector._
+import freeeventsourcing.ProcessAction.FirstOf.{ Alternative, Alternatives }
+import freeeventsourcing.ProcessAction._
+import freeeventsourcing.support.{ AggregateFromId, ValidAggregate }
+import freeeventsourcing.utils.StringSerializable
+import shapeless.ops.coproduct.{ Remove, Selector, Unifier }
+import shapeless.{ :+:, ::, CNil, Coproduct, HList, HNil, Inl, Inr }
+
+/** Nice monadic syntax to write processes. */
+class ProcessSyntax[BC <: BoundedContext](boundedContext: BC) {
+  type Action[+A] = ProcessAction[BC, A]
+  type ProcessMonad[A] = Free[Action, A]
+
+  type ValidSelector[S] = freeeventsourcing.support.ValidSelector[BC, S]
+
+  import Builders._
+
+  /** Await an event using a selector. The event must be from this bounded context. */
+  def await[S <: WithEventType: EventSelector: ValidSelector](selector: S) =
+    lift(AwaitEvent[BC, S](selector))
+
+  /** Send commands to a specific aggregate. */
+  def on[Id, A](aggregate: Id)(implicit afi: AggregateFromId[Id, BC#Aggregates]) = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method name does not match the regular expression '"^[a-z][A-Za-z0-9]*$"'](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2571398663/issues/source?bid=3298199&fileBranchId=3298202#l30)